File: tools/lint.ts
#!/usr/bin/env -S deno run --allow-run
// Copyright 2020-2021 the Deno authors. All rights reserved. MIT license.
const release = Deno.args.includes("--release");
console.log("clippy");

const mode = release ? ["--release"] : [];
const clippy = [
  "clippy",
  "--all-targets",
  "--all-features",
  ...mode,
  "--locked",
  "--",
  "-D",
  "clippy::all",
];

const p1 = new Deno.Command("cargo", {
  args: clippy,
  stdin: "null",
});

const o1 = await p1.output();

if (o1.code !== 0) {
  throw new Error(`Failed: ${clippy.join(" ")}`);
}

console.log("deno lint");

const cargoTargetDir = Deno.env.get("CARGO_TARGET_DIR") || "./target";
const dlint = `${cargoTargetDir}/${
  release ? "release" : "debug"
}/examples/dlint`;
const p2 = new Deno.Command(dlint, {
  args: ["run", "benchmarks/benchmarks.ts"],
  stdin: "null",
});

const o2 = await p2.output();

if (o2.code !== 0) {
  throw new Error(`Failed: ${dlint} benchmarks/benchmarks.ts`);
}

File: tools/lint.py
#!/usr/bin/env python
# Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.

import os
import sys
import argparse
from util import enable_ansi_colors, root_path, run


def main():
    enable_ansi_colors()
    os.chdir(root_path)

    clippy()
    dlint()


def clippy():
    print "clippy"
    args = ["cargo", "clippy", "--all-targets", "--release", "--locked"]
    run(args + ["--", "-D", "clippy::all"], shell=False, quiet=True)


def dlint():
    print "deno lint"
    run(["target/release/examples/dlint", "benchmarks/benchmarks.ts"],
        shell=False, quiet=True)


if __name__ == "__main__":
    sys.exit(main())
